torch/_export/non_strict_utils.py

```diff
@@ -140,11 +141,7 @@ def make_fake_inputs(
     if context is not None:
         # This occurs when we are exporting within dynamo. There already exists
         # a toplevel TracingContext with a fake mode, so we do not want to
-        # create another fake mode. In this scenario, we also shouldn't have any
-        # constraints since the toplevel tracing context should handle it.
-        assert (
-            len(constraints) == 0
-        ), "Found constraints when tracing with a toplevel tracing context."
+        # create another fake mode.
         fake_mode = context.fake_mode
```

Comment:
cc @angelayi

Reply:
No idea what's up with the previous comment here. 

If this assertion passed before, then `dynamic_shapes` should have been empty, no? So that means that we will always now create a full `dynamic_shapes` for inputs, and thus constraints may not be empty: in particular, everything will be explicitly static. That's why we need to remove this assertion, yeah?

But what's worrying me about the comment is that, what would break if the constraints are no longer empty, is there some kind of "shadowing" done by "toplevel tracing context"?

Reply:
Yeahh I don't know what's going on here, but @angelayi said it should be fine to remove this.

It was breaking for this test case, where we torch compile with export as backend?